include/oc_cloud.h

```diff
@@ -106,6 +106,8 @@ typedef struct oc_cloud_context_t
   oc_link_t *rd_published_resources; /**< Resource links already published */
   oc_link_t *rd_delete_resources;    /**< Resource links to delete */
   bool rd_delete_all;
+  bool rd_block_republish_resources; /**< Block republishing of resources from
```

Comment:
Hm. What do you think if we changed `OC_CLOUD_INITIALIZED` to a bit so we change all `ctx->store.status == OC_CLOUD_INITIALIZED` to `ctx->store.status & OC_CLOUD_INITIALIZED` . And `OC_CLOUD_DONT_PUBLISH_RESOURCES` we will be set in response callback of publishing, and the flag is removed when:
- start cloud manager
- reconnect occurs
- any error

Reply:
Checking `ctx->store.status == OC_CLOUD_INITIALIZED` is equivalent to checking that no other flag is set. If we change it to `1` and use `&` then we will get very different results, for example:

```
ctx->store.status = 5

ctx->store.status == 1 -> false

but 

(ctx->store.status & 1) != 0 -> true
```

But that's not hard to fix, we can just create a function like this

```
/**
 * @brief Check if flag is set in given cloud status value
 *
 * @param status the cloud status value to check
 * @param flags flag(s) to check
 * @param ignore flag(s) to ignore
 * @return true   if flags are set in status
 *         false  otherwise
 */
inline bool
check_cloud_status(uint8_t status, oc_cloud_status_t flags,
                   oc_cloud_status_t ignore)
{
  status &= ~ignore;
  if (flag == OC_CLOUD_INITIALIZED) {
    return status == OC_CLOUD_INITIALIZED;
  }
  return (status & flags) != 0;
}
```

and use it instead of `==`.

Reply:
So ok, but what about:

> And OC_CLOUD_DONT_PUBLISH_RESOURCES we will be set in response callback of publishing, and the flag is removed when:
start cloud manager
reconnect occurs
any error

Reply:
That should be alright. It'll also filter-out additional duplicate publish resources requests, since `publish_resources` is invoked from 3 places:

- `oc_cloud_add_resource`
- `publish_published_resources`
  - `cloud_rd_manager_status_changed`
  - `oc_cloud_publish_resources`

`oc_cloud_add_resource` and `oc_cloud_publish_resources` are global functions; `cloud_rd_manager_status_changed` is invoked on cloud status change. So successful calls of the global functions would block republishing of resources in `cloud_rd_manager_status_changed`, which I think is alright, since the actual list of devices has been published previously, so this is more optimal than my implementation which blocks republishing only after refresh.

As far the flag removal, start of cloud manager and reconnect make sense, but which errors do you mean? Errors that occur in publish calls, or in all cloud_status handlers, or some different set? 

There is also an `oc_cloud_logout` function which is invoked in some tests. How strong is the connection between logging in and republishing of resources? Should logging out and logging in back republish resources?

Reply:
Reworked to handle `rd_publish_resources` and `rd_published_resources` lists differently.